platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -0,0 +1,60 @@
+#import "NSExpression+MGLAdditions.h"
+
+class FilterExpressionEvaluator {
+public:
+    id operator()(mbgl::Value value) {
```

Comment:
The implementation of FilterExpressionEvaluator would look identical to PropertyValueEvaluator in MGLFeature.mm. We should make that class accessible to the whole project, then reusing it instead of reinventing the wheel here.